app/controllers/api/v1/customers_controller.rb

```diff
@@ -112,11 +113,11 @@ def transform_address!(attributes, from, to)
         end
 
         if address[:state_name].present?
-          address[:state] = Spree::State.find_by(name: address[:state_name])
+          address[:state] = Spree::State.find_by(name: address[:state_name][:name])
         end
 
         if address[:country].present?
-          address[:country] = Spree::Country.find_by(name: address[:country])
+          address[:country] = Spree::Country.find_by(name: address[:country][:name])
```

Comment:
Great, this keeps the lookup by name. But I think that the intention was to be able to update the country by code, too.